python/pylibcugraph/pylibcugraph/heterogeneous_biased_neighbor_sample.pyx

```diff
@@ -0,0 +1,424 @@
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    assert_CAI_type,
+    assert_AI_type,
+    get_c_type_from_numpy_type,
+)
+from pylibcugraph.internal_types.sampling_result cimport (
+    SamplingResult,
+)
+from pylibcugraph._cugraph_c.random cimport (
+    cugraph_rng_state_t
+)
+from pylibcugraph.random cimport (
+    CuGraphRandomState
+)
+import warnings
+
+# TODO accept cupy/numpy random state in addition to raw seed.
+def heterogeneous_biased_neighbor_sample(ResourceHandle resource_handle,
+                                         _GPUGraph input_graph,
+                                         start_vertex_list,
+                                         start_vertex_offsets,
+                                         h_fan_out,
+                                         num_edge_types,
+                                         bool_t with_replacement,
+                                         bool_t do_expensive_check,
+                                         with_edge_properties=False,
+                                         prior_sources_behavior=None,
+                                         deduplicate_sources=False,
+                                         return_hops=False,
+                                         renumber=False,
+                                         retain_seeds=False,
+                                         compression='COO',
+                                         compress_per_hop=False,
+                                         random_state=None,
+                                         return_dict=False,):
+    """
+    # FIXME: Deprecate biased_neighbor_sample
+    Performs biased neighborhood sampling, which samples nodes from
+    a graph based on the current node's neighbors, with a corresponding fan_out
+    value at each hop. The edges are sampled with biases.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resources needed for
+        referencing data and running algorithms.
+
+    input_graph : SGGraph or MGGraph
+        The input graph, for either Single or Multi-GPU operations.
+
+    edge_biases: FIXE: update this - Create edge_biases of type
+        'cugraph_edge_property_view_t' - edge biases not yet supported.
+
+    start_vertex_list: device array type
+        Device array containing the list of starting vertices for sampling.
+
+    start_vertex_offsets: list[int] (Optional)
+        Offsets of each label within the start vertex list.
+
+    h_fan_out: tuple of numpy array type
+        Device array containing the branching out (fan-out) degrees per
+        starting vertex for each hop level
+
+        The sampling method can use different fan_out values for each edge type
+        which is not the case for homogeneous neighborhood sampling.
+
+    num_edge_types: Number of edge types where a value of 1 translates to homogeneous neighbor
```

Comment:
I think this one might need a better explanation, and/or an example below. Also, shouldn't it have the type (eg. list, int, etc.) after the colon like the other args do?

Reply:
I updated the docstring to capture the relationship between the `fan_out` and the `num_edge_types`.